src/mixins/createFormLabelMixin.ts

```diff
@@ -125,92 +115,84 @@ const labelDefaults = {
 	hidden: false
 };
 
-const createFormLabelMixin: FormLabelMixinFactory = createEvented.mixin({
-	mixin: {
-		getFormFieldNodeAttributes(this: FormLabel): VNodeProperties {
-			const { properties, type } = this;
-			const attributeKeys = Object.keys(properties);
+/**
+ * Allowed attributes for a11y
+ */
+const allowedAttributes = ['checked', 'describedBy', 'disabled', 'invalid', 'maxLength', 'minLength', 'multiple', 'name', 'placeholder', 'readOnly', 'required', 'type', 'value'];
 
-			if (type) {
-				attributeKeys.push('type');
-			}
+function getFormFieldA11yAttributes(instance: FormLabel) {
+	const { properties, type } = instance;
+	const attributeKeys = Object.keys(properties);
 
-			const allowedAttributes = ['checked', 'describedBy', 'disabled', 'invalid', 'maxLength', 'minLength', 'multiple', 'name', 'placeholder', 'readOnly', 'required', 'type', 'value'];
-			const nodeAttributes: any = {};
+	if (type) {
+		attributeKeys.push('type');
+	}
 
-			for (const key of allowedAttributes) {
+	const nodeAttributes: any = {};
 
-				if (attributeKeys.indexOf(key) === -1) {
-					continue;
-				}
-				else if (key === 'type') {
-					nodeAttributes.type = type;
-				}
-				else if (key === 'readOnly' && properties.readOnly) {
-					nodeAttributes.readonly = 'readonly';
-					nodeAttributes['aria-readonly'] = true;
-				}
-				else if (key === 'invalid') {
-					nodeAttributes['aria-invalid'] = properties.invalid;
-				}
-				else if (key === 'describedBy') {
-					nodeAttributes['aria-describedby'] = properties.describedBy;
-				}
-				else if ((key === 'maxLength' || key === 'minLength' || key === 'checked') && typeof properties[key] !== 'string') {
-					nodeAttributes[key.toLowerCase()] = '' + properties[key];
-				}
-				else {
-					nodeAttributes[key.toLowerCase()] = properties[key];
-				}
-			}
+	for (const key of allowedAttributes) {
 
-			return nodeAttributes;
+		if (attributeKeys.indexOf(key) === -1) {
+			continue;
 		}
-	},
-
-	initialize(instance: any) {
-		instance.own(instance.on('properties:changed', (evt: PropertiesChangeEvent<FormLabelMixin, FormLabelMixinProperties>) => {
-			instance.tagName = evt.properties.label ? 'label' : 'div';
-		}));
-
-		if (instance.properties && instance.properties.label) {
-			instance.tagName = 'label';
+		else if (key === 'type') {
```

Comment:
follow up, I am not sure a `switch` would be correct given the logic in the `if else` conditions.